### PR TITLE
added get and get_all method in transaction class/6557

### PR DESCRIPTION
**Pull request by HemangChothani**
issue [6557]

### CODE REVIEW

File: firestore/google/cloud/firestore_v1/transaction.py [hunk 1]
```diff
@@ -200,6 +202,37 @@ def _commit(self):
...
+            query, or :data:`None` if the document does not exist.
+        """
+        if isinstance(ref_or_query, DocumentReference):
+            return self._client.get_all([ref_or_query], transaction=self._id)
+        elif isinstance(ref_or_query, Query):
+            return ref_or_query.stream(self._id)
+        else:
+            raise ValueError(
+                'Value for argument "refOrQuery" must be a DocumentReference or a Query.'
+            )
```
Comment (mf2199) [inline firestore/google/cloud/firestore_v1/transaction.py:234]: This should better be `Value for argument "ref_or_query"...`

File: firestore/tests/unit/v1/test_transaction.py [hunk 2]
```diff
@@ -329,6 +329,109 @@ def test__commit_failure(self):
...
+        # Attach the fake GAPIC to a real client.
+        client._firestore_api_internal = firestore_api
+
+    def test_get_all(self):
+        from google.cloud.firestore_v1.document import DocumentSnapshot
+        import types
+
+        client = _make_client()
+        trans = self._make_one(client)
+        data1 = {"a": u"cheese"}
```
Comment (mf2199) [inline firestore/tests/unit/v1/test_transaction.py]: Better use `transaction = self._make_one(client)`, to make it consistent with the other test methods.

File: firestore/tests/unit/v1/test_transaction.py [hunk 3]
```diff
@@ -329,6 +329,109 @@ def test__commit_failure(self):
...
+        document_pb2, read_time = _doc_get_info(document2._document_path, data2)
+        response1 = _make_batch_response(found=document_pb1, read_time=read_time)
+        response2 = _make_batch_response(found=document_pb2, read_time=read_time)
+
+        self._get_all_helper(client, [response1, response2])
+        # Actually call get_all().
+        snapshots = trans.get_all([document1, document2])
+        self.assertIsInstance(snapshots, types.GeneratorType)
+        snapshots = list(snapshots)
+        snapshot1 = snapshots[0]
```
Comment (mf2199) [inline firestore/tests/unit/v1/test_transaction.py]: I'd rather name it `snapshot0 = snapshots[0]`, `snapshot1 = snapshots[1]` etc., to make it more intuitive.

File: firestore/tests/unit/v1/test_transaction.py [hunk 4]
```diff
@@ -329,6 +329,109 @@ def test__commit_failure(self):
...
+        self.assertIsInstance(snapshot2, DocumentSnapshot)
+        self.assertIs(snapshot2._reference, document2)
+        self.assertEqual(snapshot2._data, data2)
+
+    def test_get_document_ref(self):
+        from google.cloud.firestore_v1.document import DocumentSnapshot
+        import types
+
+        client = _make_client()
+        trans = self._make_one(client)
```
Comment (mf2199) [inline firestore/tests/unit/v1/test_transaction.py]: Same as previously.

File: firestore/tests/unit/v1/test_transaction.py [hunk 5]
```diff
@@ -329,6 +329,109 @@ def test__commit_failure(self):
...
+        data1 = {"a": u"cheese"}
+        document1 = client.document("pineapple", "lamp1")
+        document_pb1, read_time = _doc_get_info(document1._document_path, data1)
+        response1 = _make_batch_response(found=document_pb1, read_time=read_time)
+        self._get_all_helper(client, [response1])
+        snapshots = trans.get(document1)
+        self.assertIsInstance(snapshots, types.GeneratorType)
+        snapshots = list(snapshots)
+        snapshot1 = snapshots[0]
```
Comment (mf2199) [inline firestore/tests/unit/v1/test_transaction.py]: Same as previously.

File: firestore/tests/unit/v1/test_transaction.py [hunk 6]
```diff
@@ -329,6 +329,109 @@ def test__commit_failure(self):
...
+        firestore_api.run_query.assert_called_once_with(
+            parent_path,
+            query._to_protobuf(),
+            transaction=trans._id,
+            metadata=client._rpc_metadata,
+        )
+
+    def test_get_failure(self):
+        client = _make_client()
+        trans = self._make_one(client)
```
Comment (mf2199) [inline firestore/tests/unit/v1/test_transaction.py]: Same as previously.

File: firestore/tests/unit/v1/test_transaction.py (hunk 3 above)
Comment (HemangChothani) [reply firestore/tests/unit/v1/test_transaction.py]: ok, I will do that if you want, but the same thing "snapshot1 = snapshots[0]" like is in the test_client.py file in the same method def test_get_all(), so  to uniform everything I use it,please verify it, if you want  i will change that variable name

Comment (mf2199) [reply firestore/tests/unit/v1/test_transaction.py]: This is not critical, was merely a suggestion. You can do it either way.

File: firestore/google/cloud/firestore_v1/transaction.py [hunk 7]
```diff
@@ -200,6 +202,37 @@ def _commit(self):
...
+            .DocumentSnapshot: The next document snapshot that fulfills the
+            query, or :data:`None` if the document does not exist.
+        """
+        return self._client.get_all(references, transaction=self._id)
+
+    def get(self, ref_or_query):
+        """
+        Retrieve a document or a query result from the database.
+        Args:
+            ref_or_query The document references or query object to return.
```
Comment (sumit-ql) [inline firestore/google/cloud/firestore_v1/transaction.py:222]: Please add param type like below: 
ref_or_query 
                (:class:`~google.cloud.firestore_v1.document.DocumentReference` or 
                :class:`~google.cloud.firestore_v1.query.Query`):
                The document references or query object to return.

File: firestore/google/cloud/firestore_v1/transaction.py (hunk 1 above)
Comment (sumit-ql) [reply firestore/google/cloud/firestore_v1/transaction.py:234]: I think, we should raise a TypeError, not ValueError here:
raise TypeError('ref_or_query" must be of type  DocumentReference or a Query.')

File: firestore/google/cloud/firestore_v1/transaction.py (hunk 7 above)
Comment (HemangChothani) [reply firestore/google/cloud/firestore_v1/transaction.py:222]: actually in the project prepared a  doc with two patterns one is declare `type and :param` and other what i use here that you can  in the whole file , so to match that pattern i use this format, if i change that than may be need to change for whole file.

File: firestore/google/cloud/firestore_v1/transaction.py (hunk 1 above)
Comment (HemangChothani) [reply firestore/google/cloud/firestore_v1/transaction.py:234]: ok i will do that

File: firestore/google/cloud/firestore_v1/transaction.py (hunk 7 above)
Comment (sumit-ql) [reply firestore/google/cloud/firestore_v1/transaction.py:222]: Ok, go ahead with what you have added.

**Review by mf2199 — APPROVED** [review]

**Pull request closed without merging**